frontend/src/components/EditModal.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import { Modal, Select } from 'antd';
+import _ from 'lodash';
+import axios from 'axios';
+
+import { api_endpoints } from '../helpers/constants'
+
+const { Option } = Select;
+
+class EditModal extends Component {
+
+  state = {
+    headquarterId: 1
+  }
+
+  componentDidUpdate(prevProps) {
+    const { offices } = this.props;
+
+    if (!_.isEqual(offices, prevProps.offices) && !!offices.length) {
+      this.setState({
+        headquarterId: !!offices.find(office => office.headquarter) ? offices.find(office => office.headquarter).id : offices[0].id
```

Comment:
Not good here, if there is matching office, then it will do same query once again.

Reply:
I need to be more detail oriented...
So 
```
const office = offices.fine(office => office.headquarter )
this.setState({
    headquarterId: !!office ? office.id: offices[0].id
})
```
It should work.